ui/src/pages/version/components/forms/components/transformer/components/TransformationGraph.js

```diff
@@ -0,0 +1,356 @@
+            column.jsonPath.startsWith(MODEL_RESPONSE_PREFIX)
+          ) {
+            links.push({ source: modelServiceNode.id, target: id });
+          }
+        });
+    }
+  });
+};
+
+const addVariablesInputNodesLinks = (
+  nodes,
+  links,
+  nodeMap,
+  idx,
+  variables,
+  stage
+) => {
+  const id = `${stage}-variables-input-${idx}`;
+  let vars = [];
+
+  variables.forEach(variable => {
+    if (variable.name !== undefined && variable.name !== "") {
+      vars.push(variable.name);
+      nodeMap[variable.name] = id;
+    }
+
+    if (
+      stage === "postprocess" &&
+      variable.jsonPath !== undefined &&
+      variable.jsonPath.startsWith(MODEL_RESPONSE_PREFIX)
+    ) {
+      links.push({ source: modelServiceNode.id, target: id });
+    }
+  });
+
+  nodes.push({
+    id: id,
+    label: `#${idx + 1} - Variables\nOutput: ${vars.join(", ")}`
+  });
+};
+
+const addTableTransformationNodesLinks = (
+  nodes,
+  links,
+  nodeMap,
+  idx,
+  tableTransformation,
+  stage
+) => {
+  const id = `${stage}-table-transformation-${idx}`;
+  const outputTable = tableTransformation.outputTable;
+  const label = `#${idx + 1} - Table Transformation\nOutput: ${outputTable}`;
+
+  const sourceTable = tableTransformation.inputTable;
+  if (nodeMap.hasOwnProperty(sourceTable)) {
+    links.push({ source: nodeMap[sourceTable], target: id });
+  }
+
+  tableTransformation.steps &&
+    tableTransformation.steps.forEach(step => {
+      step.updateColumns &&
+        step.updateColumns.forEach(updateColumn => {
+          if (
+            updateColumn.expression !== undefined &&
+            updateColumn.expression !== ""
+          ) {
+            if (nodeMap.hasOwnProperty(updateColumn.expression)) {
+              links.push({
+                source: nodeMap[updateColumn.expression],
+                target: id
+              });
+            }
+          }
+        });
+    });
+
+  nodes.push({ id: id, label: label });
+  nodeMap[outputTable] = id;
+};
+
+const addTableJoinNodesLinks = (
+  nodes,
+  links,
+  nodeMap,
+  idx,
+  tableJoin,
+  stage
+) => {
+  const id = `${stage}-table-join-${idx}`;
+  const outputTable = tableJoin.outputTable ? tableJoin.outputTable : "";
+  const label = `#${idx + 1} - Table Join\nOutput: ${outputTable}`;
+
+  const leftTable = tableJoin.leftTable;
+  if (nodeMap.hasOwnProperty(leftTable)) {
+    links.push({ source: nodeMap[leftTable], target: id });
+  }
+
+  const rightTable = tableJoin.rightTable;
+  if (nodeMap.hasOwnProperty(rightTable)) {
+    links.push({ source: nodeMap[rightTable], target: id });
+  }
+
+  nodes.push({ id: id, label: label });
+  nodeMap[outputTable] = id;
+};
+
+const addFieldLinks = (links, nodeMap, id, field, stage) => {
+  if (field.fields && field.fields.length > 0) {
+    field.fields.forEach(f => {
+      addFieldLinks(links, nodeMap, id, f, stage);
+    });
+  }
+
+  if (field.fromTable) {
+    if (field.fromTable.tableName && field.fromTable.tableName !== "") {
```

Comment:
why not `if (field.fromTable && field.fromTable.tableName && field.fromTable.tableName !== "")`

Reply:
to mitigate nested conditional